synapseclient/__main__.py

```diff
@@ -134,7 +134,71 @@ def sync(args, syn):
                                retries=args.retries)
 
 
+def _recursive_store(args, syn):
+    # Credit for the original design of this function goes to Larsson Omberg
+    # https://gist.github.com/larssono/db35917cf58440fe0b19
+
+    # Determine and confirm local root directory
+    args.file, args.FILE = args.FILE if args.FILE else args.file, None
+    if not os.path.isdir(args.file):
+        raise ValueError('Only folders can be recursively uploaded.')
+
+    # Confirm that target entity on Synapse is a folder
+    if args.parentid is None:
+        raise ValueError('--parentId is required for recursive uploads.')
+    parent = syn.get(args.parentid, downloadFile=False)
+    parent_type = parent.concreteType.split(".")[-1]
+    if not parent_type == "Folder" and not parent_type == "Project":
+        raise ValueError('--parentId must refer to a folder or project.')
+
+    # Some CLI arguments are ignored with --recursive; set to None
+    ignored = ["name", "annotations", "replace", "used", "executed"]
```

Comment:
Should `id ` be added?

Reply:
It could. Because `--parentId` and `--id` are mutually exclusive, I wouldn't expect `id` to be present past the conditional on line 147. That said, I'll add it because it's useful for developers to see the expectation. 

```suggestion
    ignored = ["id", "name", "annotations", "replace", "used", "executed"]
```